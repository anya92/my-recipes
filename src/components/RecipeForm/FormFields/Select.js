import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Label from '../styled/Label';

const StyledSelect = styled.div`
  margin: 10px 0;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  max-width: 100%;
  ${Label} {
    margin: 0 10px;
  }
  select {
    width: 50%;
    background-color: #F2F2F2;
    border: none;
    cursor: pointer;
    &:focus {
      outline: none;
    }
    ${props => props.error && 'border: 2px solid crimson;'};    
  }
`;

const Select = ({ category, handleInputChange, error }) => (
  <StyledSelect error={error}>
    <Label htmlFor="category">Category *</Label>
    <select
      id="category"
      value={category || 'Choose'}
      onChange={e => handleInputChange(e, 'category')}
    >
      <option disabled hidden>Choose</option>
      <option value="breakfast">Breakfast</option>
      <option value="dinner">Dinner</option>
      <option value="dessert">Dessert</option>
      <option value="snacks">Snack</option>
      <option value="drinks">Drink</option>
    </select>
  </StyledSelect>
);

Select.propTypes = {
  category: PropTypes.string.isRequired,
  handleInputChange: PropTypes.func.isRequired,
  error: PropTypes.bool.isRequired,
};

export default Select;
