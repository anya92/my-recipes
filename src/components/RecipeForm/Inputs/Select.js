import React from 'react';
import styled from 'styled-components';

const StyledSelect = styled.div`
  margin: 10px 0;
  font-size: 1.2rem;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  max-width: 100%;
  label {
    color: #4F4F4F;
    font-weight: 700;
    margin: 0 10px;
  }
  select {
    width: 50%;
    background-color: #F2F2F2;
    border: none;
    cursor: pointer;
    &:focus {
      outline: none;
    }
    ${props => props.error && 'border: 2px solid crimson;'};    
  }
`;

const Select = ({ category, handleInputChange, error }) => (
  <StyledSelect error={error}>
    <label htmlFor="category">Category</label>
    <select
      id="category"
      value={category || 'Choose'}
      onChange={e => handleInputChange(e, 'category')}
    >
      <option disabled hidden>Choose</option>
      <option value="breakfast">Breakfast</option>
      <option value="dinner">Dinner</option>
      <option value="dessert">Dessert</option>
      <option value="snack">Snack</option>
      <option value="drink">Drink</option>
    </select>
  </StyledSelect>
);

export default Select;
