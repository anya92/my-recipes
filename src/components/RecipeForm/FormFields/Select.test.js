import React from 'react';
import { shallow } from 'enzyme';

import Select from './Select';

let category;
let handleInputChange;
let error;
let wrapper;

beforeEach(() => {
  category = '';
  handleInputChange = jest.fn();
  error = false;
  wrapper = shallow(<Select
    category={category}
    handleInputChange={handleInputChange}
    error={error}
  />);
});

test('should render Select with no selected category correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render Select with selected category correctly', () => {
  category = 'breakfast';
  wrapper = shallow(<Select
    category={category}
    handleInputChange={handleInputChange}
    error={error}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should call handleInputChange on select change', () => {
  const value = 'breakfast';
  wrapper.find('select').simulate('change', {
    target: { value },
  });
  expect(handleInputChange).toHaveBeenLastCalledWith(
    expect.objectContaining({
      target: expect.any(String),
    }),
    'category',
  );
});
