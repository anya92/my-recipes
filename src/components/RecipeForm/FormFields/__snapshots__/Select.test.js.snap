// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Select with no selected category correctly 1`] = `
<styled.div
  error={false}
>
  <styled.label
    htmlFor="category"
  >
    Category *
  </styled.label>
  <select
    id="category"
    onChange={[Function]}
    value="Choose"
  >
    <option
      disabled={true}
      hidden={true}
    >
      Choose
    </option>
    <option
      value="breakfast"
    >
      Breakfast
    </option>
    <option
      value="dinner"
    >
      Dinner
    </option>
    <option
      value="dessert"
    >
      Dessert
    </option>
    <option
      value="snacks"
    >
      Snack
    </option>
    <option
      value="drinks"
    >
      Drink
    </option>
  </select>
</styled.div>
`;

exports[`should render Select with selected category correctly 1`] = `
<styled.div
  error={false}
>
  <styled.label
    htmlFor="category"
  >
    Category *
  </styled.label>
  <select
    id="category"
    onChange={[Function]}
    value="breakfast"
  >
    <option
      disabled={true}
      hidden={true}
    >
      Choose
    </option>
    <option
      value="breakfast"
    >
      Breakfast
    </option>
    <option
      value="dinner"
    >
      Dinner
    </option>
    <option
      value="dessert"
    >
      Dessert
    </option>
    <option
      value="snacks"
    >
      Snack
    </option>
    <option
      value="drinks"
    >
      Drink
    </option>
  </select>
</styled.div>
`;
